### ISSUE TITLE
Automation of galasactl CLI integration testing

### ISSUE THREAD

**Issue by techcobweb** [galasa-dev/projectmanagement#1413]
## Story
As a user of the galasactl tools I want to be sure that it works before I download and try it out.

## Background
We need some automation to test that the galasactl command works in many of it's combinations of
commands and parameters.

So far we have a weedy set of tests which run after the unit tests on a local build, that's not good enough to spot many things which could break, despite the code coverage climbing to 66% overall now.

We need a test suite of some sort which gets executed in the CLI pipeline which tries out the tool.

For example: create a test, build it, run it locally, push it to artifactory, run it on an ecosystem, repeat this for maven and gradle building. Run in linux and zlinux? 

Create a manual test suite we can run on windows also. Should be the same tests, but deploying and kicking things off on Windows may be more of an issue.

## Tasks
- [x] Create a simple test suite we can embellish later to run. eg: Do something like we do in the build-locally.sh script ?
- [x] Decide whether this uses bash or go
- [x] ~Work out how to push a failing and working test to galasaprod so we can call them.~ Moved to new work item as extended scope.
- [x] test-locally script written and merged into CLI repo
- [x] test-ecosystem script written and merged into CLI repo
- [x] Launch the test suite within the CLI pipeline

**Comment by jadecarino** [comment]
PR opened with test scripts for local and ecosystem test submits. PR to add execution of these test scripts into the CLI pipelines almost there just need to figure out which image to run these scripts in (must be able to execute bash, java, maven and gradle). 
Pushing generated projects and running those in an ecosystem moved to item with extended scope.

**Comment by jadecarino** [comment]
PRs: 
- https://github.com/galasa-dev/cli/pull/71 (Needs to be merged first)
- https://github.com/galasa-dev/automation/pull/196
